<launch>
    <group ns="pointcloud_processing">
        <node pkg="pointpkg" type="pointer" name="pointer_node">

            <param name="MAIN_rate" value="100"/> <!--loop rate-->
			<param name="MAIN_ShowTime" value="0"/>


			<param name="COMMON_robot_type" value="1"/> <!--1:Cobotta 2:MSRobot-->


			<param name="MANAGER_ShowTime" value="0"/>

			<param name="GUI_enable_GUI" value="1"/>

			<!--ROSのトピックを通じてポイントクラウド取得する際に使う-->
			<param name="ROS_delay" value="0.001"/> <!--time delay-->
            <param name="ROS_param_txt" value="C:/Users/y4236/Documents/pcws/src/pointpkg/ros_interface_parameters_MSR.txt"/>
			<param name="ROS_use_color" value="1"/>


			<param name="PCL_pcd_type" value="4"/><!--多分使ってない-->
			<!--contact detectorで使うポイントクラウド3種-->
            <param name="PCL_pcd_target" value="C:\Users\y4236\Documents\pcws\src\pointpkg\pcd\CobottaCommon\CobottaToolTip_blue_target.pcd"/>
            <param name="PCL_pcd_mask" value="C:\Users\y4236\Documents\pcws\src\pointpkg\pcd\CobottaCommon\CobottaToolTip_blue_mask.pcd"/>
            <param name="PCL_pcd_tip" value="C:\Users\y4236\Documents\pcws\src\pointpkg\pcd\CobottaCommon\CobottaToolTip_blue_tip.pcd"/>
            <param name="PCL_param_txt" value="C:/Users/y4236/Documents/pcws/src/pointpkg/pcl_interface_parameters.txt"/><!--座標変換などのパラメータ保存-->
            <param name="PCL_pcd_read" value="C:\Users\y4236\Documents\pcws\src\pointpkg\pcd\CobottaCommon\CobottaToolTip_target.pcd"/><!--予め保存したポイントクラウドの場所-->
            <param name="PCL_pcd_write" value="C:/Users/y4236/Documents/pcws/src/pointpkg/pcd/23_02_05_graph_45_4/"/><!--保存するポイントクラウドの場所-->
			<param name="PCL_YOLO_RGB_YG" value="255000000"/><!--３桁ごとにRGB-->
			<param name="PCL_YOLO_RGB_BP" value="255000"/>
			<param name="PCL_CalibrationYOLO" value="0"/><!--旧キャリブレーション-->
			<param name="PCL_ShowTipPosition0" value="0"/>　<!--座標を出力する．黄緑-->
			<param name="PCL_ShowTipPosition1" value="0"/>
			<param name="PCL_ShowMarkersMeanPosition0" value="0"/><!--マーカー平均点を表示する -->
			<param name="PCL_ShowMarkersMeanPosition1" value="0"/>
			<param name="PCL_CalibrateRotZ" value="0"/><!--vertical推奨 -->
			<param name="PCL_ShowCoordinate" value="0"/><!--描画の際，座標軸表示する-->
			<param name="PCL_Show200Frames" value="0"/><!--200フレーム出力．修論実験用-->
			<!--描画の際の背景色-->
			<param name="PCL_BG_R" value="0.6"/>
			<param name="PCL_BG_G" value="0.6"/>
			<param name="PCL_BG_B" value="0.6"/>
			<param name="PCL_CalibrateO" value="0"/><!--ツール先端をz軸の0になるように-->


			<param name="PCL_enable_PCL" value="1"/>
			<param name="PCL_enable_ReadPCD" value="0"/>
			<param name="PCL_enable_WritePCD" value="0"/>
			<param name="PCL_enable_Transform" value="1"/>
			<param name="PCL_enable_PassThrough" value="1"/>
			<param name="PCL_enable_ParticleFilter" value="0"/>
			<param name="PCL_enable_DetectWithSim" value="0"/>
			<param name="PCL_enable_Recognite" value="0"/>
			<param name="PCL_enable_RemovePlane" value="0"/><!--最大平面除去-->
			<param name="PCL_enable_DrawResult" value="1"/><!--ポイントクラウド描画-->
			<param name="PCL_enable_DetectWithYOLO" value="1"/>
			<param name="PCL_enable_CalibrationHorizon" value="0"/><!--大きな水平面の物体おいて使う-->
			<param name="PCL_enable_CalibrationVertical" value="0"/><!--大きな鉛直面の物体おいて使う-->

			<param name="CD_DownSampleTargetCloud" value="0"/> <!--表示するツールの色薄くてもよいか？-->
			<param name="CD_ShowConsensus" value="1"/> <!--ツールの推定位置とかぶっているポイントクラウドを移すかどうか-->
			<param name="CD_ShowTargetTool" value="0"/> <!--ツールの推定位置姿勢表示-->
			<param name="CD_ShowContactPoints" value="0"/> <!--探索点回りのポイント着色-->
			<param name="CD_KeepTip" value="0"/><!-- t:0 先端取り除かないなら1.もう使ってない-->
			<param name="CD_SearchMode" value="1"/><!--0:探索しない 1:vector 2:multiset-->
			

			<param name="SIM_enable_SIM" value="0"/><!--CoppeliaSimと通信必要？-->
			<param name="SIM_simIP" value="127.0.0.1"/>
            <param name="SIM_PortNumber" value="19998"/>
            <param name="SIM_param_txt" value="C:/Users/y4236/Documents/pcws/src/pointpkg/coppeliasim_interface_parameters.txt"/>
            <param name="SIM_LabView_connect" value="0"/><!--LabViewから姿勢取得用-->
            <param name="SIM_LabView_IP" value="192.168.1.100"/>
            <param name="SIM_LabView_Port" value="6430"/>

			<param name="RS_enable_RS" value="1"/>
			<param name="RS_enable_FILTER_Decimate" value="0"/><!--RSInterface::set_filter参照-->
			<param name="RS_enable_FILTER_Disparity" value="1"/>
			<param name="RS_enable_FILTER_Spatial" value="0"/>
			<param name="RS_enable_FILTER_Temporal" value="1"/>
			<param name="RS_enable_FILTER_Hole" value="1"/>
			<param name="RS_TYoverYG" value="0.038"/><!--(先端-黄色マーカー)と（黄-緑）の長さ比-->
			<param name="RS_TBoverBP" value="1.40"/><!--青赤バージョン-->
			<param name="RS_Length_TC" value="0.0035"/><!--0.0035,0.0012 ツールの軸半径-->
			<param name="RS_ZratioLimit" value="1.05"/><!--背景分離閾値-->
			<param name="RS_updateLastZ" value="0"/><!--背景分離閾値を動的に変化させるか．使わないほうが良いとき多そう-->
			<param name="RS_ShowTime" value="0"/><!--処理にかかった時間表示-->
			<param name="RS_ShowImage" value="1"/><!--RGB，抽出結果，デプス表示-->
			<param name="RS_SavePictures" value="0"/><!--同保存-->
			<param name="RS_PicturesFolder" value="C:/Users/y4236/Desktop/Pictures/"/><!--SavePicture有効時，保存場所を指定する-->
			<param name="RS_ShowInference" value="0"/><!--YOLOの結果を表示-->
			<param name="RS_ShowMarkers" value="1"/><!--マーカーのポイントを表示させるか-->
			<param name="RS_EnableThread" value="0"/><!--並列処理用．高速化できるけど，標準出力とかは表示されなかったりするかも-->
			<param name="RS_EnableMarkerPoseCompensater" value="0"/><!--機械学習でマーカー位置から先端位置を補正するクラス．データ集められず実質機能してない-->
			<param name="RS_EnableCompensateMarkersDistance" value="0"/><!--色や姿勢から先端位置をマニュアルで補正しようとしたやつ．使わないほうがよさそう-->

			<!--
			<param name="YOLO_ModelPath" value="C:/Users/y4236/Documents/ToolDetectYOLO/yolov7/CobottaTool/result/train8/weights/last.onnx"/>
			-->
			<param name="YOLO_ModelPath" value="C:/Users/y4236/Documents/ToolDetectYOLO/yolov7/CobottaTool/result/train20/weights/best.onnx"/> <!--YOLOの重みデータのパス-->
			<param name="YOLO_enable_GPU" value="1"/> <!--GPU or CPU-->
			<param name="YOLO_Conf" value="0.3"/><!--quiitaや論文等参考-->
			<param name="YOLO_IoU" value="0.6"/><!--quiitaや論文等参考-->

			<param name="MPC_ModelPath" value="C:/Users/y4236/Documents/pcws/src/pointpkg/model.onnx"/><!--MarkerPositionCompensater用のモデル保存場所-->



			<!-- <rosparam command="load" file="$(find robot_solver)/config/CobottaDH.yaml"/> -->
        </node>
    </group>
</launch>